sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala

```diff
@@ -78,4 +78,68 @@ class QueryParsingErrorsSuite extends QueryTest with SharedSparkSession {
         message = "Invalid SQL syntax: LATERAL can only be used with subquery.")
     }
   }
+  test("MORE_THAN_ONE_FROM_TO_UNIT_IN_INTERVAL_LITERAL: from-to unit in the interval literal") {
+    val e = intercept[ParseException] {
+      spark.sql("SELECT INTERVAL 1 to 3 year to month AS col")
+    }
+    assert(e.getErrorClass === "MORE_THAN_ONE_FROM_TO_UNIT_IN_INTERVAL_LITERAL")
+    assert(e.getMessage.contains(
+      "Can only have a single from-to unit in the interval literal syntax"))
+  }
+
+  test("INVALID_INTERVAL_LITERAL: invalid interval literal") {
+    val e = intercept[ParseException] {
+      spark.sql("SELECT INTERVAL DAY")
+    }
+    assert(e.getErrorClass === "INVALID_INTERVAL_LITERAL")
+    assert(e.getMessage.contains(
+      "at least one time unit should be given for interval literal"))
+  }
+
+  test("INVALID_FROM_TO_UNIT_VALUE: value of from-to unit must be a string") {
+    val e = intercept[ParseException] {
+      spark.sql("SELECT INTERVAL -2021 YEAR TO MONTH")
+    }
+    assert(e.getErrorClass === "INVALID_FROM_TO_UNIT_VALUE")
+    assert(e.getMessage.contains(
+      "The value of from-to unit must be a string"))
+  }
+
+  test("UNSUPPORTED_FEATURE: Unsupported from-to interval") {
+    val e = intercept[ParseException] {
+      spark.sql("SELECT extract(MONTH FROM INTERVAL '2021-11' YEAR TO DAY)")
+    }
+    assert(e.getErrorClass === "UNSUPPORTED_FEATURE")
+    assert(e.getSqlState === "0A000")
+    assert(e.getMessage.contains("The feature is not supported"))
```

Comment:
Could you check whole error message, please.

Reply:
@MaxGekk Changed as per review comments.